src/widgetastic_patternfly/__init__.py

```diff
@@ -1804,3 +1805,140 @@ def items(self):
             # value will include the label from the <strong> block, parse it out
             items.update({key: element_text.replace(key, '', 1).lstrip()})
         return items
+
+
+class DatePicker(Widget):
+    """Represents the Bootstrap DatePicker.
+
+      Args:
+        name: name of DatePicker
+        id: id of DatePicker
+
+    .. code-block:: python
+
+       date = DatePicker(name='miq_date_1')
+    """
+    ROOT = ParametrizedLocator(".//*[contains(@class, 'datepicker-dropdown')]")
+    TEXT_BOX = ".//*[@name='{name}' or @id='{id}']"
+    DATE = ".//*[contains(@class, 'datepicker-days')]/table/tbody/tr/*"
+    MONTH = ".//*[contains(@class, 'datepicker-months')]/table/tbody/tr/td/*"
+    YEAR = ".//*[contains(@class, 'datepicker-years')]/table/tbody/tr/td/*"
+    PREV = ".//*[contains(@class, 'prev')]"
+    NEXT = ".//*[contains(@class, 'next')]"
+    DATE_PICKER_SWITH = ".//*[contains(@class, 'datepicker-switch')]"
+    DATE_MAPPING = {'dd': '%d', 'mm': '%m', 'MM': '%B', 'yy': '%y', 'yyyy': '%Y'}
+
+    def __init__(self, parent, name=None, id=None, logger=None):
+        """Create the widget"""
+        Widget.__init__(self, parent, logger=logger)
+        self.name = name
+        self.id = id
+        self.TEXT_BOX = self.TEXT_BOX.format(name=name, id=id)
+
+    @property
+    def value(self):
+        return self.parent_browser.get_attribute('value', self.TEXT_BOX)
```

Comment:
REQUIRED: good job ! however, I think you have to turn your xpaths above into separate widgets in this widget/view.
Your widget is too dynamic.